### PR TITLE
normalize-name-in-interface-and-type-alias

### LINKED ISSUE #2350: enum name is not normalizated in modular but in hlc

### LINKED ISSUE #2482: SDK Generator using Typespec - Casing is incorrect for union
For the following TypeSpec:

```
union recordingChannelType {
  /** Mixed channel type */
  "Mixed",

  /** Unmixed channel type */
  "Unmixed",

  string,
}
```

The Generated Typescript look like:

```
export type recordingChannelType = string;
```

But, the corresponding swagger definition which looks like the following:

```
"recordingChannelType": {
      "type": "string",
      "description": "Recording channel type",
      "enum": [
        "Mixed",
        "Unmixed"
      ],
      "x-ms-enum": {
        "name": "recordingChannelType",
        "modelAsString": true,
        "values": [
          {
            "name": "Mixed",
            "value": "Mixed",
            "description": "Mixed channel type"
          },
          {
            "name": "Unmixed",
            "value": "Unmixed",
            "description": "Unmixed channel type"
          }
        ]
      }
    }
```

The Generated Typescript looks like the following:

```
export type RecordingChannelType = string;
```

i.e. Look at the casing at the beginning. The generator must be fixed to correct this issue.

@xirzec @lmazuel @joheredi FYI.....

### PR DESCRIPTION
**Pull request by qiaozha**
fixes #2350 
fixes #2482

### CODE REVIEW

**Comment by qiaozha** [comment]
> Should we also normalize the model name here https://github.com/Azure/autorest.typescript/blob/main/packages/typespec-test/test/hierarchy_generic/generated/typespec-ts/src/models/models.ts#L12 ? RLC has done that https://github.com/Azure/autorest.typescript/blob/main/packages/typespec-test/test/hierarchy_generic/generated/typespec-ts/src/rest/models.ts#L12
> 
> @joheredi, any suggestions?

I don't think we should block on this for now, as we have another issue to track this https://github.com/Azure/autorest.typescript/issues/2086

File: packages/typespec-ts/src/modular/buildCodeModel.ts [hunk 1]
```diff
@@ -1108,9 +1109,10 @@ function emitEnum(context: SdkContext, type: Enum): Record<string, any> {
 
   return {
     type: "enum",
-    name: getLibraryName(context, type)
-      ? getLibraryName(context, type)
-      : type.name,
+    name: normalizeName(
```
Comment (MaryGao) [inline packages/typespec-ts/src/modular/buildCodeModel.ts:1112]: do we need to do this for RLC?

File: packages/typespec-ts/test/modularUnit/enumUnion.spec.ts [hunk 2]
```diff
@@ -842,13 +842,13 @@ describe("model type", () => {
         modelFile!.getFullText()!,
         `
         export interface Test {
-          color: LR | UD;
+          color: Lr | Ud;
```
Comment (MaryGao) [inline packages/typespec-ts/test/modularUnit/enumUnion.spec.ts:845]: Could you add a UT here for name `createdByType` both for RLC and Modular?

File: packages/typespec-ts/src/modular/buildCodeModel.ts (hunk 1 above)
Comment (qiaozha) [reply packages/typespec-ts/src/modular/buildCodeModel.ts:1112]: done

File: packages/typespec-ts/test/modularUnit/enumUnion.spec.ts (hunk 2 above)
Comment (qiaozha) [reply packages/typespec-ts/test/modularUnit/enumUnion.spec.ts:845]: added

File: packages/typespec-ts/test/modularUnit/enumUnion.spec.ts [hunk 3]
```diff
@@ -842,17 +842,48 @@ describe("model type", () => {
...
         /** */
-        export type LR = "left" | "right";
+        export type Lr = "left" | "right";
         /** */
-        export type UD = "up" | "down";
+        export type Ud = "up" | "down";
         `
       );
     });
+    it("non standard enum name", async () => {
```
Comment (MaryGao) [inline packages/typespec-ts/test/modularUnit/enumUnion.spec.ts:855]:
```suggestion
    it("non-standard enum name", async () => {
```

**Review by MaryGao — APPROVED** [review]
LGTM

**Review by lirenhe — APPROVED** [review]

**Pull request merged**